### ISSUE TITLE
Can't edit ink file with Unity open

### ISSUE THREAD

**Issue by twheinen** [inkle/ink#87]
I'm using the Ink Unity Integration, and I am unable to edit and save my .ink file while Unity is open. The auto-compilation feature is great, but it kind of ruins the convenience if I have to close Unity every time I want to edit the script. Even when I disable the auto-compilation and json handling in the Ink settings, it still won't let me save the ink file with Unity open. It just keeps saying the file is open in another program. I'm not sure what's causing this behind the scenes, but it would be awesome if saving with Unity open was possible. Thanks!

**Comment by twheinen** [comment]
After further testing, I figured out that it only prevents me from saving when the ink file is selected in the Unity inspector and/or the ink file was the last selected object in the inspector. Deselecting the ink script in the inspector view won't allow you to save, but selecting another object in the scene will. Now that I know what's causing it, this issue isn't as big of an annoyance, but I guess it's up to you whether the issue should be considered closed.

**Comment by joethephish** [comment]
Thanks for the report! Definitely worth investigating to see if there's something we can do about it.

**Comment by joethephish** [comment]
Oh, important question - are you on Windows or Mac?

**Comment by tomkail** [comment]
Interesting - I'd imagine it's because we're reading the contents of the file in the inspector - something I should optimise anyway - I've just push a hotfix you can try. Let us know if it works!

**Comment by twheinen** [comment]
@joethephish I'm on Windows 10.

@tomkail I tried the hotfix, and the issue seems mostly cleared. I can save when I have the ink file open in the inspector, but there still seems to be times when I'm unable to save. It's not exactly predictable, but it mainly seems to happen right after coming out of play mode while the ink file is selected in the inspector.

Thanks for the quick response!

**Comment by tomkail** [comment]
Interesting! Do you have a slow computer, or a large ink file? I'd imagine
that it's because you can't save while its reading, but you'd expect that
to be very fast.
Also - which editor are you using for ink?

**Comment by twheinen** [comment]
My computer is pretty fast, and the ink file I'm testing is tiny. I've been using Notepad++ to edit the file. I can do more testing tomorrow and try and narrow down the cause.

**Comment by tomkail** [comment]
That's good to know, thanks. That would be massively appreciated, let us
know how it goes!

**Comment by Fahrengeit** [comment]
Yeah, I had this issue too. First of all, I think it appeared with new update of Unity, because I didn't notice this before, always had a window of Ink Player opened. And I've tested it now - on 5.3.1 everything works great. But with 5.3.4 (maybe even 5.3.3 started this problem) I've found this issue.
It appears always after pressing Play in Ink Player. After that you can not save the file. Workaround - Play another ink file, so Unity can "release" ink document, but it's not so convenient as it was before.

**Comment by tomkail** [comment]
Thanks, that's really good to know. I'll take a look later today and see
what we can do.

**Comment by Horatio-Blackwood** [comment]
Just to chime in with another report, I've seen this as well with Windows 7 and both sublime text and notepad++ as my ink editor.

**Comment by joaorequiao** [comment]
I'm having the same problem on Windows 10. I'm using Inky to edit the files. The file is really small (6 lines) and my pc is very fast (Core i7 , 8Gb RAM).  If I open a random script after opening the ink file, it all goes well.
